docs/sources/reference/stdlib/map.md

```diff
@@ -0,0 +1,35 @@
+---
+canonical: https://grafana.com/docs/alloy/latest/reference/stdlib/map/
+description: Learn about map functions
+aliases:
+  - ./concat/ # /docs/alloy/latest/reference/stdlib/concat/
+menuTitle: map
+title: map
+---
+
+# map
```

Comment:
Technically, this page should be called "object" and the function should be `object.inner_join`. That's because the [Alloy syntax type](https://grafana.com/docs/alloy/latest/get-started/configuration-syntax/expressions/types_and_values/#objects) is called "object". I wonder if we should rename the Alloy syntax type?
* "Map" would ring more bells in people's minds than "object"
* We already document this as "map" in component reference docs
* "Object" makes me think of the sort of thing a "capsule" does

Reply:
I realised that even "map" is not an appropriate namespace for this function, since it operates on arrays of maps. I changed the function name to `targets.merge`.